File: Server/src/main/content/global/skill/combat/summoning/familiar/npc/SpiritDagannothNPC.java
package content.global.skill.combat.summoning.familiar.npc;

import content.global.skill.combat.summoning.familiar.Familiar;
import content.global.skill.combat.summoning.familiar.FamiliarSpecial;
import core.game.node.entity.combat.equipment.WeaponInterface;
import core.game.node.entity.player.Player;
import core.plugin.Initializable;

@Initializable
public class SpiritDagannothNPC extends Familiar {

    public SpiritDagannothNPC() {
        this(null, 6804);
    }

    public SpiritDagannothNPC(Player owner, int id) {
        super(owner, id, 5700, 12017, 6, WeaponInterface.STYLE_CONTROLLED);
    }

    @Override
    public Familiar construct(Player owner, int id) {
        return new SpiritDagannothNPC(owner, id);
    }

    @Override
    protected boolean specialMove(FamiliarSpecial special) {
        return false;
    }

    @Override
    public int[] getIds() {
        return new int[]{6804, 6805};
    }

}

File: Server/src/main/content/global/skill/combat/summoning/familiar/npc/SpiritDagannothNPC.kt
package content.global.skill.combat.summoning.familiar.npc

import content.global.skill.combat.summoning.familiar.Familiar
import content.global.skill.combat.summoning.familiar.FamiliarSpecial
import core.game.node.entity.combat.equipment.WeaponInterface
import core.game.node.entity.player.Player
import core.plugin.Initializable

/**
 * Spirit dagannoth familiar.
 */
@Initializable
class SpiritDagannothNPC @JvmOverloads constructor(owner: Player? = null, id: Int = 6804) : Familiar(owner, id, 5700, 12017, 6, WeaponInterface.STYLE_CONTROLLED) {

    override fun construct(owner: Player, id: Int): Familiar {
        return SpiritDagannothNPC(owner, id)
    }

    override fun specialMove(special: FamiliarSpecial): Boolean {
        return false
    }

    override fun getIds(): IntArray {
        return intArrayOf(6804, 6805)
    }
}
